packages/sdk/src/assets/bridge-asset.ts

```diff
@@ -0,0 +1,267 @@
+
+      const bridgeRequestPayloadStub = this._createBridgeOutPayload(
+        chainId,
+        holographToChainId,
+        contractAddress,
+        maxGasValue,
+        maxGasValue,
+        bridgeOutPayload,
+      )
+
+      const consumedGas = BigInt(
+        (
+          await this._operator.simulateContractFunction({
+            chainId,
+            functionName: 'jobEstimator',
+            args: [bridgeRequestPayloadStub],
+            options: {gas: gasLimitStub}, //{account: gasEstimationHelperAddress , gas: TEST_GAS_LIMIT},
+          })
+        ).result,
+      )
+
+      destinationGasLimit = gasLimitStub - consumedGas //TODO: ver se dá para chamar o gasEstimate ao invés de fazer essa conta
+    }
+
+    if (gasController?.gasLimitMultiplier) {
+      destinationGasLimit = (destinationGasLimit * BigInt(gasController.gasLimitMultiplier)) / BigInt(100)
+    }
+
+    if (gasController?.gasPrice) {
+      destinationGasPrice = gasController.gasPrice
+    } else {
+      // we extract the most recent and most accurate gas prices for destination network
+      const provider = this._providers.byChainId(chainId)
+      const gasPricing: GasPricing = await initializeGasPricing(provider)
+
+      // get gas price based on eip-1559 support
+      destinationGasPrice = gasPricing.isEip1559 ? gasPricing.maxFeePerGas! : gasPricing.gasPrice!
+    }
+
+    if (gasController?.gasPriceMultiplier) {
+      destinationGasPrice = (destinationGasPrice * BigInt(gasController.gasPriceMultiplier!)) / BigInt(100)
+    }
+
+    return {
+      destinationGasPrice,
+      destinationGasLimit,
+    }
+  }
+
+  async estimateBridgeOut(
+    chainId: number,
+    toChainId: number,
+    contractAddress: Address,
+    bridgeOutPayload: Hex,
+  ): Promise<EstimateBridgeOutResult> {
+    const {destinationGasPrice, destinationGasLimit} = await this._estimateBridgeOutDestinationGas(
+      chainId,
+      toChainId,
+      contractAddress,
+      bridgeOutPayload,
+    )
+
+    const holographToChainId = getNetworkByChainId(toChainId).holographId
+
+    // now that we have gasLimit and gasPrice, we update the payload to include the proper data needed for estimating fees
+    const bridgeRequestPayload = (
+      await this._createBridgeOutPayload(
+        chainId,
+        holographToChainId,
+        contractAddress,
+        destinationGasLimit,
+        destinationGasPrice,
+        bridgeOutPayload,
+      )
+    ).result
+
+    // we are now ready to get fees for transaction
+    const feeValue = await this._getValue(
+      chainId,
+      holographToChainId,
+      destinationGasLimit,
+      destinationGasPrice,
+      bridgeRequestPayload,
+    )
+
+    const unsignedBridgeOutTx = BridgeAsset.createUnsignedBridgeOutTx(
+      holographToChainId,
+      contractAddress,
+      destinationGasLimit,
+      destinationGasPrice,
+      bridgeOutPayload,
+      //{value: feeValue}, TODO: do we need to pass the value here?
```

Comment:
I believe it should be here. As this should generate the entire tx payload, except that its not singed and submitted.